mp/src/game/shared/momentum/mom_stickybomb.cpp

```diff
@@ -0,0 +1,357 @@
+
+    // Functions
+    DEFINE_ENTITYFUNC(Touch),
+END_DATADESC();
+#endif
+
+IMPLEMENT_NETWORKCLASS_ALIASED(MomStickybomb, DT_MomStickybomb)
+
+BEGIN_NETWORK_TABLE(CMomStickybomb, DT_MomStickybomb)
+#ifdef CLIENT_DLL
+RecvPropVector(RECVINFO(m_vInitialVelocity)),
+RecvPropInt(RECVINFO(m_bTouched)),
+RecvPropVector(RECVINFO_NAME(m_vecNetworkOrigin, m_vecOrigin)),
+RecvPropEHandle(RECVINFO(m_hLauncher)),
+#else
+SendPropVector(SENDINFO(m_vInitialVelocity), 20, 0, -3000, 3000),
+SendPropExclude("DT_BaseEntity", "m_vecOrigin"),
+SendPropBool(SENDINFO(m_bTouched)),
+SendPropVector(SENDINFO(m_vecOrigin), -1, SPROP_COORD_MP_INTEGRAL | SPROP_CHANGES_OFTEN, 0.0f, HIGH_DEFAULT, SendProxy_Origin),
+SendPropEHandle(SENDINFO(m_hLauncher)),
+#endif
+END_NETWORK_TABLE();
+
+LINK_ENTITY_TO_CLASS(momentum_stickybomb, CMomStickybomb);
+PRECACHE_WEAPON_REGISTER(momentum_stickybomb);
+
+#ifdef CLIENT_DLL
+static MAKE_CONVAR(mom_sj_stickybomb_drawdelay, "0.1", FCVAR_ARCHIVE | FCVAR_CLIENTDLL,
+                   "Determines how long it takes for stickies to start being drawn upon spawning.\n", 0, 1);
+#endif
+
+CMomStickybomb::CMomStickybomb()
+{
+    m_vInitialVelocity.Init();
+    m_bTouched = false;
+    m_flChargeTime = 0.0f;
+#ifdef CLIENT_DLL
+    m_flSpawnTime = 0.0f;
+#else
+    m_flDamage = 0.0f;
+    m_flCreationTime = 0.0f;
+    m_bUseImpactNormal = false;
+    m_vecImpactNormal.Init();
+#endif
+}
+
+CMomStickybomb::~CMomStickybomb()
+{
+}
+
+void CMomStickybomb::UpdateOnRemove()
+{
+    // Tell our launcher that we were removed
+    CMomentumStickybombLauncher *pLauncher = dynamic_cast<CMomentumStickybombLauncher *>(m_hLauncher.Get());
+
+    if (pLauncher)
+    {
+        pLauncher->DeathNotice(this);
+    }
+
+    BaseClass::UpdateOnRemove();
+}
+
+void CMomStickybomb::Spawn()
+{
+    BaseClass::Spawn();
+#ifdef CLIENT_DLL
+    m_flSpawnTime = gpGlobals->curtime;
+#else
+    SetModel(MOM_STICKYBOMB_MODEL);
+    BaseClass::Spawn();
+
+    UseClientSideAnimation();
+    SetCollisionGroup(COLLISION_GROUP_PROJECTILE);
+    SetSolidFlags(FSOLID_NOT_STANDABLE);
+    SetMoveType(MOVETYPE_FLYGRAVITY, MOVECOLLIDE_FLY_BOUNCE);
+    SetSolid(SOLID_BBOX);
+    AddEffects(EF_NOSHADOW);
+    SetSize(Vector(-2, -2, -2), Vector(2, 2, 2));
+    AddFlag(FL_GRENADE);
+
+    VPhysicsInitNormal(SOLID_BBOX, 0, false);
+
+    m_bTouched = false;
+    m_flCreationTime = gpGlobals->curtime;
+    m_takedamage = DAMAGE_NO;
+    SetGravity(MOM_STICKYBOMB_GRAVITY);
+    SetFriction(MOM_STICKYBOMB_FRICTION);
+    SetElasticity(MOM_STICKYBOMB_ELASTICITY);
+#endif
+}
+
+void CMomStickybomb::Precache()
+{
+    BaseClass::Precache();
+
+    PrecacheModel(MOM_STICKYBOMB_MODEL);
+}
+
+#ifdef CLIENT_DLL
+
+void CMomStickybomb::OnDataChanged(DataUpdateType_t type)
+{
+    BaseClass::OnDataChanged(type);
+
+    if (type == DATA_UPDATE_CREATED)
+    {
+        m_flCreationTime = gpGlobals->curtime;
+        ParticleProp()->Create("stickybombtrail_red", PATTACH_ABSORIGIN_FOLLOW);
```

Comment:
You should be using the weapon script for the particles, see how the rocket trail particle is created for reference.